scripts/redirect.js

```diff
@@ -0,0 +1,21 @@
+/* eslint-disable no-restricted-globals */
+/* eslint-disable prefer-regex-literals */
+if (new RegExp('hlx\\.live$').test(location.hostname) || new RegExp('hlx\\.page$').test(location.hostname)) {
+  const queryParams = new URLSearchParams(location.search);
+  const newQp = new URLSearchParams();
+  const qpList = ['domainkey', 'startdate', 'enddate', 'url'];
+  let requiredQueryParamsFlag = true;
+  qpList.forEach((qp) => {
+    if (!queryParams.has(qp) || queryParams.get(qp) === null || queryParams.get(qp) === '') {
+      requiredQueryParamsFlag = false;
+    } else {
+      newQp.set(qp, queryParams.get(qp));
+    }
```

Comment:
I'd prioritize positive check and simplify the check itself, isolating it.
```suggestion
    const hasValidQueryParam = queryParams.has(qp) && queryParams.get(qp) !== null && queryParams.get(qp).length > 0
    if (hasValidQueryParam) {
      newQp.set(qp, queryParams.get(qp));
    } else {
      requiredQueryParamsFlag = false;
    }
```